app/v1/controllers/requestParsers/validators/DeleteCodingOutValidator.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers.validators
+
+import v1.controllers.requestParsers.validators.validations.{NinoValidation, TaxYearValidation}
+import v1.models.errors.MtdError
+import v1.models.request.deleteCodingOut.DeleteCodingOutRawRequest
+
+class DeleteCodingOutValidator extends Validator[DeleteCodingOutRawRequest] {
+
+  private val validationSet = List(parameterFormatValidation)
+
+  private def parameterFormatValidation: DeleteCodingOutRawRequest => List[List[MtdError]] = { data =>
+    List(
+      NinoValidation.validate(data.nino),
+      TaxYearValidation.validate(data.taxYear)
+    )
```

Comment:
The current file you're using to validate the tax year uses the wrong error and doesn't validate for:

`RULE_TAX_YEAR_NOT_SUPPORTED`
`RULE_TAX_YEAR_RANGE_INVALID`